### PR TITLE
refactor: remove unused doorway uic components

### PR DESCRIPTION
**Pull request by emilyjablonski**
This PR addresses https://github.com/bloom-housing/Bloom-Doorway/issues/27

- [ ] Addresses the issue in full
- [x] Addresses only certain aspects of the issue

Related to the dependency issues

## Description

Removes unused `doorway-ui-components`. I checked out `Find all references` for each component. The vast majority weren't used anywhere, so I deleted the file. Another good portion were used, but the files were identical to UIC, so I just pulled in the UIC version and deleted the file. For any components that were used that had Doorway customizations, even if they were really small, I just left them in to tackle later and kept the file.

As a result of some of the CSS being removed, there were a handful of places I needed to update classes in the public site, but it was just around shifting semibold --> bold to maintain the current design. The other change was an a11y fix that was flagged by a linter.

## How Can This Be Tested/Reviewed?

The only place there were slight changes was in the components that we use on all the help / information pages under `Help Center` and `Partners`, so I would check those out and compare the styles to prod.

## Author Checklist:

- [ ] Added QA notes to the issue with applicable URLs
- [x] Reviewed in a desktop view
- [x] Reviewed in a mobile view
- [x] Reviewed considering accessibility
- [ ] Added tests covering the changes
- [ ] Made corresponding changes to the documentation
- [ ] Ran `yarn generate:client` and/or created a migration when required

## Review Process:

- Read and understand the issue
- Ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Either (1) explicitly ask a clarifying question, (2) request changes, or (3) approve the PR, even if there are very small remaining changes, if you don't need to re-review after the updates

### CODE REVIEW

File: doorway-ui-components/index.ts [hunk 1]
```diff
@@ -1,164 +1,50 @@
 /* Actions */
```
Comment (emilyjablonski) [inline doorway-ui-components/index.ts:1]: A good overview of what was removed

File: sites/public/src/components/shared/DoorwayCollapsibleSection.tsx [hunk 2]
```diff
@@ -19,12 +19,19 @@ const DoorwayCollapsibleSection = ({
 
   return (
     <div
-      className={`doorway-collapsible-section ${rootClassNames}`}
+      className={`${styles["doorway-collapsible-section"]} ${rootClassNames}`}
       onClick={() => {
         setExpanded(!isExpanded)
       }}
+      role="button"
```
Comment (emilyjablonski) [inline sites/public/src/components/shared/DoorwayCollapsibleSection.tsx:26]: This was an existing a11y problem, caught by a linter bc this file moved

**Review by ColinBuyck — APPROVED** [review]
LGTM just noticed something really subtle with the collapsible sections. Based on the video below, it seems like the focus state of single line titles have different widths and on multi-line titles, the arrow itself shifts over which doesn't occur on prod

https://github.com/user-attachments/assets/7252a75f-9e8b-4cb2-92be-3bca61a7077f

**Review by ludtkemorgan — APPROVED** [review]
🧹

**Comment by emilyjablonski** [comment]
@ColinBuyck updated, ty for catching that! was relying on a margin class that only existed in doorway-ui-components

**Pull request merged**